File: airesources/Python/BasicBot.py
from hlt import *
from networking import *

myID, gameMap = getInit()
sendInit("BasicPythonBot")

while True:
	moves = []
	gameMap = getFrame()

	for y in range(gameMap.height):
		for x in range(gameMap.width):
			site = gameMap.getSite(Location(x, y))
			if site.owner == myID:
				direction = random.randint(0, 5)
				if site.strength < 5*site.production:
					direction = STILL
				else:
					for d in CARDINALS:
						if gameMap.getSite(Location(x, y), d).owner != myID:
							direction = d
							break
				moves.append(Move(Location(x, y), direction))

	sendFrame(moves)

File: airesources/Java/BasicBot.java
import java.util.ArrayList;

public class BasicBot {
	public static void main(String[] args) {
		InitPackage iPackage = Networking.getInit();
		int myID = iPackage.myID;
		GameMap gameMap = iPackage.map;

		Networking.sendInit("BasicJavaBot");

		while(true) {
			ArrayList<Move> moves = new ArrayList<Move>();

			gameMap = Networking.getFrame();

			for(int y = 0; y < gameMap.height; y++) {
				for(int x = 0; x < gameMap.width; x++) {
					Site site = gameMap.getSite(new Location(x, y));
					if(site.owner == myID) {
						Direction moveDirection = Direction.randomDirection();
						if(site.strength < site.production*5) {
							moveDirection = Direction.STILL;
						} else {
							for(Direction d : Direction.CARDINALS) {
								if(gameMap.getSite(new Location(x, y), d).owner != myID) {
									moveDirection = d;
									break;
								}
							}
						}
						moves.add(new Move(new Location(x, y), moveDirection));
					}
				}
			}

			Networking.sendFrame(moves);
		}
	}
}
